coresdk/src/backend/database_driver.cpp

```diff
@@ -79,21 +80,21 @@ bool sk_query_success(sk_query_result result)
     return result._result == SQLITE_ROW || result._result == SQLITE_OK || result._result == SQLITE_DONE;
 }
 
-bool sk_query_get_next_row(sk_query_result result)
+int sk_rows_affected(sk_database db)
 {
-    if (result._result == SQLITE_ROW)
-    {
-        sqlite3_stmt *stmt = sqlite3_stmt_from_void(result._stmt);
-        result._result = sqlite3_step(stmt);
-        return true;
-    }
-    raise_warning("Attempt to read a row from a query without data");
-    return false;
+    return sqlite3_changes(sqlite3_from_void(db._data));
```

Comment:
I suggest that you call this inside of `run_sql` and store the result in the `sk_query_result`. Then the student can ask at any time "how many rows were affected by this result?" and this will work even if they execute additional commands.